File: frontend/src/types/message.ts
export type MessageType = "TEXT" | "IMAGE" | "FILE" | "SYSTEM";

export interface Message {
  id: string;
  conversationId: string;
  senderId: string;
  senderUsername: string;
  type: MessageType;
  content: string;
  replyToMessageId: string | null;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface SendMessageRequest {
  conversationId: string;
  content: string;
  type?: MessageType;
  replyToMessageId?: string | null;
}

File: backend/src/main/java/com/trieuhuy/chatapp/domain/model/Message.java
package com.trieuhuy.chatapp.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Message {
    private UUID id;
    private UUID conversationId;
    private UUID senderId;
    private String type; // TEXT | IMAGE | FILE | SYSTEM
    private String content;
    private UUID replyToMessageId;
    private boolean isDeleted;
    private Instant createdAt;
    private Instant updatedAt;
}

